src/foobar/rest/serializers/purchase.py

```diff
@@ -33,5 +35,40 @@ def as_purchase_kwargs(self):
         }
 
 
+class PurchaseItemSerializer(serializers.Serializer):
+    qty = serializers.IntegerField()
+    product_id = serializers.CharField()
+
+
 class PurchaseSerializer(serializers.Serializer):
+    """Serializes a purchase, as retrieved from the foobar API call:
+        `get_purchase()`
+    """
+    id = serializers.UUIDField(allow_null=False)
     amount = MoneyField()
+    status = IntEnumField()
+
+    def to_representation(self, instance):
+        purchase_obj = super().to_representation(instance[0])
+        item_serializer = PurchaseItemSerializer(instance[1], many=True)
+        purchase_obj['items'] = item_serializer.data
+        return purchase_obj
+
+
+class PurchaseStatusSerializer(serializers.Serializer):
+    status = serializers.CharField(allow_null=False)
+
+    def validate_status(self, value):
+        if value.upper() not in PurchaseStatus.members():
+            raise serializers.ValidationError(
+                _('Status named: {} not found.'.format(value))
+            )
+        if value.upper() == 'PENDING':
```

Comment:
It could be a good idea to redo this part so that we can rely entirely on the transitions that are defined on the status enum. Always nice with single source of truth.